src/applications/vaos/covid-19-vaccine/redux/actions.js

```diff
@@ -292,30 +294,27 @@ export function getAppointmentSlots(startDate, endDate, initialFetch = false) {
       try {
         const startDateString = !fetchedStartMonth
           ? startDate
-          : moment(endDate)
-              .startOf('month')
-              .format('YYYY-MM-DD');
+          : format(startOfMonth(parseISO(endDate)), 'yyyy-MM-dd');
```

Comment:
There seems to be a lot of redundant conversions when passing arguments. `startDate` and `endDate` are passed in as strings, converted to `Date` before being formatted into strings again as arguments for `getSlots` then converted back into `Date` again before being formatted into strings again as arguments to `getAvailableV2Slots`.

Let's just create the Date object in `SelectDate1Page` and `DateTimeSelectPage/index` once and pass it as a arguments to all child functions then format into string when needed.

Reply:
Also, why are we formatting this into `yyyy-MM-dd` if all we are going to do in `getSlots` is parsing it into Date then formatting it with `toISOString`? Sounds like we can just call `toISOString` here instead.